pkg/sql/schemachanger/scbuild/dependencies.go

```diff
@@ -30,6 +30,7 @@ import (
 type Dependencies interface {
 	scbuildstmt.ClusterAndSessionInfo
 	scbuildstmt.Telemetry
+	CatalogReader
```

Comment:
Why are you extending this interface as opposed to using the accessor to it below? I think we need to pick one.